lib/fiwareDeviceSimulator.js

```diff
@@ -992,6 +1010,45 @@ function getExternalData(options, callback) {
 }
 
 /**
+ * Generates the final external URL in case it uses some date templates.
+ * @param  {String} url The external url.
+ * @return {String}     The final external url with all the date templates subtituted by their final values.
+ */
+function generateExternalURL(url) {
+  var now = new Date();
+  var finalURL = url.replace(/{YY}/g, now.getFullYear().toString().substring(2,4));
+  finalURL = finalURL.replace(/{YYYY}/g, now.getFullYear().toString());
+  var month = now.getMonth() + 1; 
+  finalURL = finalURL.replace(/{MM}/g, month < 10 ? '0' + month : month.toString());
+  finalURL = finalURL.replace(/{M\*}@en/g, MONTHS.en[now.getMonth()].toUpperCase());
+  finalURL = finalURL.replace(/{m\*}@en/g, MONTHS.en[now.getMonth()].toLowerCase());
+  finalURL = finalURL.replace(/{MMM}@en/g, MONTHS.en[now.getMonth()].substring(0,3).toUpperCase());
+  finalURL = finalURL.replace(/{mmm}@en/g, MONTHS.en[now.getMonth()].substring(0,3).toLowerCase());
+  finalURL = finalURL.replace(/{M\*}@es/g, MONTHS.es[now.getMonth()].toUpperCase());
+  finalURL = finalURL.replace(/{m\*}@es/g, MONTHS.es[now.getMonth()].toLowerCase());
+  finalURL = finalURL.replace(/{MMM}@es/g, MONTHS.es[now.getMonth()].substring(0,3).toUpperCase());
+  finalURL = finalURL.replace(/{mmm}@es/g, MONTHS.es[now.getMonth()].substring(0,3).toLowerCase());
+  var date = now.getDate();
```

Comment:
Perhaps a local var for `now.getMonth()` (or two local vars for:   `MONTHS.en[now.getMonth()]` and `MONTHS.es[now.getMonth()]`.

such a change would yield shorter lines and make it a little faster in run-time.

[ Same for DAYS below ]

Just a suggestion ...